File: Luisz576 Minecraft Plugin/src/main/java/com/luisz/luisz576api/domain/game/games/TheBridgeGame.java
package com.luisz.luisz576api.domain.game.games;

import com.luisz.luisz576api.domain.game.GameID;
import com.luisz.luisz576api.domain.game.GameModel;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.UUID;

public class TheBridgeGame extends GameModel {
    public final String mapName;
    public final TheBridgeGameMode gameMode;
    public final List<UUID> players = new ArrayList<>(), winners = new ArrayList<>();
    //todo vai ser quando acabar a partida? timestamp: Date,
    public TheBridgeGame(String mapName, TheBridgeGameMode gameMode, Collection<UUID> players, Collection<UUID> winners) {
        super(GameID.THE_BRIDGE);
        this.mapName = mapName;
        this.gameMode = gameMode;
        this.players.addAll(players);
        this.winners.addAll(winners);
    }
}

File: Api/src/schemas/games/the_bridge/TheBridgeGame.ts
import { ITheBridgeGame, ITheBridgePlayer } from "../../../domain/models/games/the_bridge/TheBridgeGame"
import { IPlayerProfile } from "../../../domain/models/player_profile/PlayerProfile"
import { TheBridgeDb } from "../../../services/database"
import { Document, Schema } from "mongoose"

const TheBridgeGame = new Schema<ITheBridgeGame>({
    timestamp: {
        type: Date,
        default: Date.now,
        immutable: true
    },
    game_mode: {
        type: Number,
        required: true
    },
    map_name: {
        type: String,
        required: true
    },
    players: {
        type: [{
            uuid: {
                type: String,
                required: true
            },
            placed_blocks: {
                type: Number,
                required: true
            },
            eaten_golden_apples: {
                type: Number,
                required: true
            },
            score: {
                type: Number,
                required: true
            },
            kills: {
                type: Number,
                required: true
            },
            deaths: {
                type: Number,
                required: true
            },
            timestamp: {
                type: Date,
                default: Date.now,
                immutable: true
            }
        }],
        required: true
    },
    winners: {
        type: [String],
        required: true
    },
})

TheBridgeGame.methods.getPlayers = async function(): Promise<IPlayerProfile[]>{
    throw new Error('Not implemented')
}

export function calculateProfileNetworkXP(player: ITheBridgePlayer, winner: boolean): number{
    // TODO
    return 0
}

export type ITheBridgeGameModel = ITheBridgeGame & Document
export default TheBridgeDb.model<ITheBridgeGameModel>('TheBridgeGame', TheBridgeGame)